### DISCUSSION TITLE
How to speed up build extensions and do it with one command?

### DISCUSSION THREAD

**Discussion by Enhed** [directus/directus#16279] [Archive]
Often I can have many extensions in each project. I collect them when I update the version of the whole image, all extensions are delivered inside the docker image. If I use a native build utility, I need to build each extension separately, for example:  

```json
 "scripts": {
    "start": "npm run build-extensions && directus bootstrap && directus start",

    "build-extensions": "npm run build-swagger & npm run build-sockets",
    "build-swagger": "cd extensions-dev/enpoints/swagger && npm run build",
    "build-sockets": "cd extensions-dev/hooks/sockets && npm run build"
  },
```

Inside each extension build command, I specify the output path to the extensions folder. If the number of extensions grows to 10+, it becomes hell. There are too many commands, and the build time increases several times.  

Before i used to use the standard `tsc` command and everything worked very fast. But now the compiled files through this command do not work.  

Is it possible to speed up the build and do it with a single command from the root of the project?

**Comment by rijkvanzanten** [comment]
That's what https://github.com/directus/directus/pull/15672 is all about! 🙂

**Reply by Enhed** [reply]
in dev its working with `"directus-extension-x":"workspace:*"` but if i use `npm install` in docker i got error `Unsupported URL Type "workspace:": workspace:*`. How to use this bundle link for production?

**Reply by paescuj** [reply]
Hmm, where does this `"directus-extension-x":"workspace:*"` comes from? Did you set that on your own? If you're using yarn / pnpm locally, you need to use this in docker as well.

**Reply by Enhed** [reply]
@paescuj yes i tried set that as inside in `package` in directus project and locally its working, but in docker i got error. I need switch `npm`  to `pnpm` for docker?

**Reply by Enhed** [reply]
@paescuj i done, with pnpm all working, thx :)
